src/lib/core/iostream.h

```diff
@@ -208,6 +215,7 @@ static inline void
 iostream_ctx_clear(struct iostream_ctx *ctx)
 {
 	ctx->mode = IOSTREAM_MODE_UNINITIALIZED;
+	ctx->ssl = NULL;
```

Comment:
Shouldn't it be `ssl_iostream_ctx_delete`? Context deletion is fine even if you have already created some `SSL` objects. The `SSL_CTX` objects and their internals are reference counted in openssl.

Reply:
Ok, I see. Still looks confusing though. Very easy to leak the context because it has no sanity checks not reference counting. Anyway, currently I don't see it leaking anywhere.